### PR TITLE
1647: Add store links

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

For the download instructions of the koblenz pass self service we need store links. in #1707 we may also create a separate app download route and need also the urls for the other projects

### Proposed changes

- add store urls for all project in buildConfig
- extract `AppStoreLinks` component to make it reusable for #1707 

### Side effects

-none

### Testing

1. Ensure `npm i` ran on root folder to have new buildConfig variables
2. Start koblenz pass go to /erstellen and create a pass.
3. Go to next and check that the store links are properly set. Note that for ios the app is not visible until first production release

### Resolved issues

Fixes: #1647

### CODE REVIEW

File: frontend/build-configs/bayern/index.ts
Comment (steffenkleinle) [inline frontend/build-configs/bayern/index.ts:113]: :upside_down_face: Optional: Personally I think it might be better to also just add the `appStoreId` to the build configs, e.g. `id1261285110` in this case.
This would
- allow us to use the appStoreId separately for e.g. an `apple-itunes-app` meta tag for websites
- allow us to set the language of the link dynamically based on the users language
- reduce the chance of copy/paste mistakes like setting the language to `ky`

However, due neither of the first two points being relevant to us atm, feel free to just keep this as it is.

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox, nice :)

File: administration/src/bp-modules/components/AppStoreLinks.tsx
Comment (f1sh1918) [reply administration/src/bp-modules/components/AppStoreLinks.tsx:7]: afaik this is the correct syntax for mui styled, check here
https://v5-0-6.mui.com/system/styled/

File: administration/src/bp-modules/components/AppStoreLinks.tsx
Comment (f1sh1918) [reply administration/src/bp-modules/components/AppStoreLinks.tsx:34]: For google they always use "Google Play öffnen" and for Apple "App Store öffnen"
This is a common thing and branded on the icons

Comment (steffenkleinle) [reply administration/src/bp-modules/components/AppStoreLinks.tsx:34]: Sorry for the misunderstanding, this comment is about variable naming, not about the user facing texts.
We have link vs url and appStore vs appleStore.

File: frontend/build-configs/bayern/index.ts
Comment (f1sh1918) [reply frontend/build-configs/bayern/index.ts:107]: So you think i should create a constant for the base appStoreLink like "https://play.google...." and then construct it with the applicationId.
Hm for me the issue is that for the app store link (apple) its constructed very different and i would need to also set a base url like `https://apps.apple.com/de/app/`
and then use the application name and id `ehrenamtskarte-bayern/id1261285110`
If i do that i would have to create a separate Id for apple and export it to be able to construct the complete appStoreLink in a common way in the `AppStoreLinks`
I don't think that its worth the effort tbh

File: frontend/build-configs/bayern/index.ts
Comment (f1sh1918) [reply frontend/build-configs/bayern/index.ts:113]: hm same as i mentioned above. the i also need this`ehrenamtskarte-bayern` separately.
I don't see a big improvement here and will keep my solution

**Comment by f1sh1918** [comment]
may i get a second review @seluianova ? :)

File: frontend/build-configs/bayern/index.ts
Comment (steffenkleinle) [reply frontend/build-configs/bayern/index.ts:107]: Fine for me. You could have a look at integreat how we do it there :)

**Comment by seluianova** [comment]
Is a 'zu' needed here?
image

**Review by seluianova — APPROVED** [review]
lgtm, just one concern regarding 'zu' (not directly related to this PR, but maybe you could fix that as well)

**Pull request merged**
